core/src/main/java/hudson/slaves/ChannelPinger.java

```diff
@@ -68,40 +107,80 @@ public void preOnline(Computer c, Channel channel, FilePath root, TaskListener l
     }
 
     public void install(Channel channel) {
-        if (pingInterval < 1) {
-            LOGGER.fine("Agent ping is disabled");
+        if (pingIntervalSeconds < 1 || pingTimeoutSeconds < 1) {
+            LOGGER.warning("Agent ping is disabled");
             return;
         }
 
         try {
-            channel.call(new SetUpRemotePing(pingInterval));
+            channel.call(new SetUpRemotePing(pingTimeoutSeconds, pingIntervalSeconds));
             LOGGER.fine("Set up a remote ping for " + channel.getName());
         } catch (Exception e) {
             LOGGER.severe("Failed to set up a ping for " + channel.getName());
         }
 
         // set up ping from both directions, so that in case of a router dropping a connection,
         // both sides can notice it and take compensation actions.
-        setUpPingForChannel(channel, pingInterval, true);
+        setUpPingForChannel(channel, pingTimeoutSeconds, pingIntervalSeconds, true);
     }
 
-    private static class SetUpRemotePing extends MasterToSlaveCallable<Void, IOException> {
+    static class SetUpRemotePing extends MasterToSlaveCallable<Void, IOException> {
         private static final long serialVersionUID = -2702219700841759872L;
-        private int pingInterval;
-        public SetUpRemotePing(int pingInterval) {
-            this.pingInterval = pingInterval;
+        private transient int pingInterval;
```

Comment:
To be compatible in cases were we have the old version of the class on the other side. I had it removed at first but then decided to go the safe way. Are we sure that this situation cannot happen?, I so, I will remove it.